manim/mobject/coordinate_systems.py

```diff
@@ -2187,7 +2187,7 @@ def __init__(
         z_axis.rotate_about_zero(angle_of_vector(self.z_normal))
         z_axis.shift(
```

Comment:
There's still a problem if the z_range doesn't include 0, this should help with that.
```suggestion
        z_axis.shift(-z_axis.number_to_point(self._origin_shift(z_axis.x_range)))
        z_axis.shift(
```